ethereum/eth/src/test/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPoolTest.java

```diff
@@ -359,7 +355,9 @@ public void shouldNotAddRemoteTransactionsThatAreInvalidAccordingToStateDependen
     givenTransactionIsValid(transaction2);
     when(transactionValidator.validate(eq(transaction1), any(Optional.class), any()))
         .thenReturn(valid());
-    when(transactionValidator.validateForSender(transaction1, null, true))
```

Comment:
this is the only one I'm still not sure of

Reply:
Ok it's because it's the _other_ validateForSender() that gets called here.
This works:
```
  @Test
  public void shouldNotAddRemoteTransactionsThatAreInvalidAccordingToStateDependentChecks() {
    givenTransactionIsValid(transaction2);
    when(transactionValidator.validate(eq(transaction1), any(Optional.class), any()))
        .thenReturn(valid());
    when(transactionValidator.validateForSender(eq(transaction1), eq(null), any(TransactionValidationParams.class)))
        .thenReturn(ValidationResult.invalid(NONCE_TOO_LOW));
    transactionPool.addRemoteTransactions(asList(transaction1, transaction2));

    assertTransactionNotPending(transaction1);
    assertTransactionPending(transaction2);
    verify(batchAddedListener).onTransactionsAdded(singleton(transaction2));
    verify(transactionValidator).validate(eq(transaction1), any(Optional.class), any());
    verify(transactionValidator).validateForSender(eq(transaction1), eq(null), any(TransactionValidationParams.class));
    verify(transactionValidator).validate(eq(transaction2), any(Optional.class), any());
    verify(transactionValidator).validateForSender(eq(transaction2), any(), any());
    verifyNoMoreInteractions(transactionValidator);
  }
```